Sources/Paywalls/Components/PaywallButtonComponent.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PaywallButtonComponent.swift
+//
+//  Created by Jay Shortway on 02/10/2024.
+//
+// swiftlint:disable missing_docs
+
+import Foundation
+
+#if PAYWALL_COMPONENTS
+
+public extension PaywallComponent {
+
+    struct ButtonComponent: PaywallComponentBase {
+
+        let type: ComponentType
+        public let stack: PaywallComponent.StackComponent
+
+        public init(
+            stack: PaywallComponent.StackComponent
+        ) {
+            self.type = .button
+            self.stack = stack
+        }
+
+    }
```

Comment:
So in my packages PRs, I have that `StackableComponent` protocol (see [here](https://github.com/RevenueCat/purchases-ios/pull/4324/files#diff-6fed0f670d26d3bdf79586129d096238c544e4022a76f86a096997b387e6764dR22-R34)) that does a similar thing to this 🤔 

But instead of the `ButtonComponent` _having_ a stack, it would make the `ButtonComponent` behave like a stack.

I know we don't `StackableComponent` in this branch yet but we should probably make sure we do the same approach for all the things 😇 

I do think this having a stack looks a lot cleaner in the code! 🫶  However, it does make the JSON data structure of the `ButtonComponent` more nesty...

```js
{
  "type": "button",
  # button properties
  "stack": {
    "type": "stack",
    "components": [
      # children components
    ]
    # all the stack properties
  }
}
```

vs


```js
{
  "type": "button",
  "components": [
      # children components
    ],
   # button properties
   # all the stack properties
}
```

Any thoughts? 😇